compiler/rustc_middle/src/ty/sty.rs

```diff
@@ -266,6 +266,11 @@ impl<'tcx> ClosureArgs<'tcx> {
     pub fn print_as_impl_trait(self) -> ty::print::PrintClosureAsImpl<'tcx> {
         ty::print::PrintClosureAsImpl { closure: self }
     }
+
+    /// Prints closure signature in form of `fn(Args) -> Output` with stripped `extern "rust-call`.
+    pub fn print_signature(self) -> ty::print::PrintClosureSignature<'tcx> {
```

Comment:
Can we not just reuse `print_as_impl_trait`? Seems like it does basically the same thing, except for being prefixed with `impl Fn` instead of `fn`. I don't expect users to particularly care, since the point is the signature.

Reply:
Yes, I believe we can reuse that if we do not have anything against prefixed `impl Fn`, it conveys the same information. However, it may be confusing for the reader of the note because in the first note, we say `expected fn pointer fn(args)`, and possibly in the second one `..has signature: impl Fn(args)`, won't the reader be confused by `fn` vs `Fn`? My reasoning for using the current output of `..has signature: fn(args)` is just for consistency with the first note, but if you think it is negligible, I will change it for the `impl` variant.

Reply:
I don't think this matters

Reply:
I've found out that it is also possible to extract the fn-like signature, without the `impl` from the closure more easily with the [`signature_unclosure`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/context/struct.TyCtxt.html#method.signature_unclosure), so this may be even better?